core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -2453,6 +2467,16 @@ object ZIO extends ZIOCompanionPlatformSpecific {
   final def foreach[R, E, A, B](in: Chunk[A])(f: A => ZIO[R, E, B]): ZIO[R, E, Chunk[B]] =
     in.mapM(f)
 
+  /**
+   * Applies the function `f` to each element of the `NonEmptyChunk[A]` and
+   * returns the results in a new `NonEmptyChunk[B]`.
+   *
+   * For a parallel version of this method, see `foreachPar`.
+   * If you do not need the results, see `foreach_` for a more efficient implementation.
+   */
+  final def foreach[R, E, A, B](in: NonEmptyChunk[A])(f: A => ZIO[R, E, B]): ZIO[R, E, NonEmptyChunk[B]] =
```

Comment:
We should eventually add variants for `foreachParN` and `collectAllParN` as well as aliases in the companion objects.

Reply:
To be sure, do you mean moving to `object Chunk`?

Reply:
No. In `ZIO.scala` there are `foreachParN` variants and `collectAllParN` variants. So we should implement versions for `NonEmptyChunk` in that file. Then all of these methods are aliased in `UIO.scala`, `URIO.scala`, `Task.scala`, etc...

Reply:
ok, I see, they don't exist yet for `Chunk[A]`:
```scala
scala> ZIO.foreachParN(5)(Chunk())
<console>:25: error: type mismatch;
 found   : zio.Chunk[Nothing]
 required: Iterable[?]
       ZIO.foreachParN(5)(Chunk())
```

I will add both versions of them.

Reply:
I did the alias for all the methods in UIO, Task, ..., ...

For `foreachParN` on Chunk and NonEmpty that will go in another PR.

Reply:
Agreed.